akka-persistence/src/main/scala/akka/persistence/journal/WriteJournalBase.scala

```diff
@@ -4,16 +4,39 @@
 
 package akka.persistence.journal
 
-import akka.persistence.{ PersistentRepr, PersistentEnvelope }
 import akka.actor.Actor
+import akka.persistence.{ Persistence, PersistentEnvelope, PersistentRepr }
+
 import scala.collection.immutable
 
 private[akka] trait WriteJournalBase {
   this: Actor ⇒
 
+  lazy val persistence = Persistence(context.system)
+  private def eventAdapters = persistence.adaptersFor(self)
+
   protected def preparePersistentBatch(rb: immutable.Seq[PersistentEnvelope]): immutable.Seq[PersistentRepr] =
-    rb.collect {
-      case p: PersistentRepr ⇒ p.update(sender = Actor.noSender) // don't store sender
+    rb.collect { // collect instead of flatMap to avoid Some allocations
+      case p: PersistentRepr ⇒
+        p.update(sender = Actor.noSender) // don't store sender
```

Comment:
it is immutable, so you must pass in this result in next stmt

    adaptToJournal(p.update(sender = Actor.noSender))